Frameworks/CoreText/CTFramesetter.mm

```diff
@@ -16,96 +16,55 @@
 
 #import <CoreText/CTFrameSetter.h>
 #import <StubReturn.h>
+#import <algorithm>
 #import "CoreTextInternal.h"
 #import "CGPathInternal.h"
+#import "DWriteWrapper.h"
+
+using namespace std;
 
 @implementation _CTFrameSetter : NSObject
 - (void)dealloc {
     [super dealloc];
 }
 @end
 
-static id _createFrame(_CTFrameSetter* frameSetter, CGRect frameSize, CGSize* sizeOut, bool createFrame) {
+static _CTFrame* _createFrame(_CTFrameSetter* frameSetter, CGRect frameSize, CGSize* sizeOut, bool createFrame) {
     _CTFrame* ret = nil;
 
     if (createFrame) {
-        ret = [_CTFrame alloc];
+        ret = [_CTFrame new];
```

Comment:
Welcome to the *ref world - all reference (CTFrameRef/_CTFrame*) are all +1. It is the responsibility of the caller to call CFRelease() on them.

The convention I have followed in the code is - if an API returns a NS* type it follows the regular NS memory management rules. If an API returns a *Ref type it is always +1, so the caller is responsible for freeing them

---
In reply to: [77270014](https://github.com/Microsoft/WinObjC/pull/950#discussion_r77270014) [](ancestors = 77270014,77269955)